### PR TITLE
add volume for the home directory to persist data 

### PR DESCRIPTION
**Pull request by yuhuyoyo**

### CODE REVIEW

File: src/shiny/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+version: '2.4'
+services:
+  app:
+    container_name: "application-server"
+    build:
+      context: .
+      dockerfile: Dockerfile
+    args:
+      - VARIANT=4
```
Comment (mbookman) [inline src/shiny/docker-compose.yaml:9]: Can you add a comment to the docker compose YAML file, or is there some place to document this configuration. How would someone maintaining this know what VARIANT 4 is and why it is set here?

**Review by mbookman — CHANGES_REQUESTED** [review]

File: src/shiny/docker-compose.yaml (hunk 1 above)
Comment (yuhuyoyo) [reply src/shiny/docker-compose.yaml]: added a comment.

i wanted to make image tag as a templatable variable so user can pick something other than the default. same for other containers. but publishing a devcontainer template to ghcr.io requires a launch and I haven't gotten around to do it.

File: src/shiny/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+services:
+  app:
+    container_name: "application-server"
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        - BASE_IMAGE=rocker/shiny  
+        # version tag of the rocker/shiny image.
+        - VARIANT=4
```
Comment (mbookman) [inline src/shiny/docker-compose.yaml:11]: A few questions on this. Why not update the Dockerfile and have `BASE_IMAGE` include the tag:

```
BASE_IMAGE=rocker/shiny:latest
```
If there's a compelling reason not to, then change this from `VARIANT` to `IMAGE_TAG` and the comment is not needed.

**Review by mbookman — CHANGES_REQUESTED** [review]

**Review by mbookman — APPROVED** [review]
LGTM!

**Pull request merged**